src/player.cpp

```diff
@@ -638,11 +649,22 @@ void Player::addStorageValue(const uint32_t key, const int32_t value, const bool
 		}
 	}
 
-	if (value != -1) {
+	if (strValue != "-1") {
```

Comment:
Dangerous. Using strings instead of integers will change the "undefined" value, "-1" should not be a special string and can be surprising. You should check for emptiness instead.